app/models/course_option.rb

```diff
@@ -0,0 +1,14 @@
+class CourseOption < ApplicationRecord
+  belongs_to :course
```

Comment:
do we actually need to perform a queries like  `course_option.course`?
maybe we do not need it, and we can avoid this line, this usually brings to less complex factories
(same reasoning for  :site, :application_choices)

Reply:
How does this affect the factories @alexdesi? I don't see a problem with specifying the relationship.

Reply:
We do use this association to do join queries, e.g. to get from `ApplicationChoice` to a `Provider`.

Reply:
Alright if we already use them.

To clarify: we do not want to mess with the model with methods that we do not need.
For example, sometimes it's easy to mess up with factory bot, I was talking about avoiding the circular dependencies issue, that happens when you try to create complex factories.
That's why I usually prefer to stay simple and create only the one-way dependency (the main one that we are going to use more often).